pkg/aws/ec2/instance.go

```diff
@@ -117,7 +118,23 @@ func (i *ec2Instance) LoadDetails(ec2APIHelper api.EC2APIHelper) error {
 		return fmt.Errorf("unsupported instance type, couldn't find ENI Limit for instance %s", i.instanceType)
 	}
 
-	i.deviceIndexes = make([]bool, limits.Interface)
+	defaultCardIdx := limits.DefaultNetworkCardIndex
+	var defaultNetworkCardLimit int64
+	for _, card := range limits.NetworkCards {
+		if card.NetworkCardIndex == int64(defaultCardIdx) {
+			defaultNetworkCardLimit = card.MaximumNetworkInterfaces
+			break
+		}
+	}
+	if defaultNetworkCardLimit == 0 {
```

Comment:
Curious, do we expect this value to be 0 ever?

Reply:
Shouldn't be. This check is for in case the limit file doesn't have cards loaded for the instance type. In the case, the `defaultNetworkCardLimit` will be default int value 0. In this case, we need to return an error.